### ISSUE TITLE
Memory getting increased when navigating one page to another?

### ISSUE THREAD

**Issue by kumaresan-subramani** [angular/angular#27980]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/....

### Is this a regression?

 Don't know

### Description

 I have created one angular project with routing feature . here i have two page each of them having grid. when i navigate one page to another dynamically, memory getting increased, i don't know why it is getting increased. Here i am doing destroy properly but memory getting increased. I could not solve this issue

## 🔬 Minimal Reproduction

 
Step 1: download sample using below link [ [http://www.syncfusion.com/downloads/support/directtrac/general/ze/naviagtinggrid-828521508]](http://www.syncfusion.com/downloads/support/directtrac/general/ze/naviagtinggrid-828521508)

Step 2: Run following commands 

>   npm install
>    ng serve

Step 3:
Then you will see application demo in  `http://localhost:4200/`. First take screenshot for memory initially.

Step 4:
 Then press start button and wait for 15 secs, then you will see **console window** show message like **`15 completed`**. Then again take screenshot for to check memory size

Step 5:
  Finally you will see memory leak

## 🔥 Exception or Error
<pre><code>

</code></pre>

## 🌍  Your Environment

**Angular Version: 7.10**
**npm Version:5.6.0**
**node Version: v8.11.3**

</code></pre>

**Anything else relevant?**

I have referred below two links and tried methods which are prescribed in those link but they wont solve this issue

https://stackoverflow.com/questions/22397484/how-to-reduce-remove-memory-leaks-in-angular-application

https://www.twilio.com/blog/prevent-memory-leaks-angular-observable-ngondestroy

**Comment by woppa684** [comment]
What kind of objects remain in memory that you didn't expect? Did you run the garbage collector before the second snapshot? Your code doesn't compile here (since it requires some external modules). The first link you mention in the end is only relevant for AngularJS (not Angular).

**Comment by ChrisDahmCS** [comment]
TLDR; I believe this issue is not an issue with Angular and if it remains an issue while using the 3rd party libs, belongs in [the syncfusion repo](https://github.com/syncfusion/ej2-angular-ui-components).

@kumaresan-subramani since your example both contains 3rd party libs and doesn't work out of the box with the instructions provided, I made a [StackBlitz example](https://stackblitz.com/edit/angular-routing-mem-example) that will likely show something similar to what you are seeing.

As @woppa684 mentioned, the GC likely wasn't run between the examples since it likely won't run immediately when there's only a small amount of work to do. The clickable example I have shows a slight increase with a few clicks, but it drops back down to the base mem amount after the GC runs (observed in Chrome dev tools -> Memory tab). Also in Chrome instead of waiting the few seconds, clicking the "collect garbage" button also reduces it back to the baseline. A side note is that the application needs to drop it's startup fluff from mem to see the increases (this may take a few seconds as well).

For me, I never got over a few kb difference in the snapshots (but of course, small components). With the 3rd party libs you are using, this becomes larger since there is a lot more code in the components. Keep in mind that when using a 3rd party lib, if you see the mem does not eventually go back down to the base amount after a few clicks (and wait a few seconds or using the collect garbage button)), it could indicate a mem leak with that lib instead. 

Also I find it's easier to try to show Angular issues with the raw Angular dependencies, not adding 3rd party dependencies that are not added with the Angular CLI as it can be less clear what the issue actually is. 

Side "tip": I use Google search and almost always use `-angularjs` in the search as well as "tools -> time" dropdown to clean up most angularjs or older results. Other search engines may have similar handy exclusion options.

**Comment by kumaresan-subramani** [comment]
I will create  new sample  with ng-grid and then i will check and update to you..!

**Comment by trotyl** [comment]
@kumaresan-subramani For Bug report you're responsible to make a minimal reproduction without any 3rd-party library, namely only `@angular/*` packages are involved.

**Comment by kumaresan-subramani** [comment]
@trotyl , OKAY
